### DISCUSSION TITLE
After Setup USB Boot

### DISCUSSION THREAD

**Discussion by Jambul46** [procount/pinn#681] [Ideas]
Would it be possible to add some "boot from that" selector to boot from "that data storage" to boot into systems that are not installed by pinn but flashed on external "data storages"?

-Sry for my english

**Comment by Jambul46** [comment]
Something like if you would connect rpi bootable usb it would automatically pop up in the boot order

**Reply by lurch** [reply]
Not possible, as PINN-installed OSes work quite differently to non-PINN-OSes; see e.g. https://github.com/raspberrypi/noobs/wiki/Standalone-partitioning-explained and https://github.com/raspberrypi/noobs/wiki/NOOBS-partitioning-explained for more detail.

**Comment by procount** [comment]
Pinn can only boot a partition on the same device as Pinn is stored on due to the firmware.
However, if you have a USB  bootable model, you could set the bootorder to boot the USB device first, so if it is plugged in, the Pi will boot it in preference to PINN on the sd card (& vuce versa). But PINN will not be able to do the selection.

**Reply by lurch** [reply]
See https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#BOOT_ORDER for more information about configuring the `BOOT_ORDER` in the EEPROM (only available on the Pi4 and Pi400).

**Reply by Jambul46** [reply]
Huh? Iam booting from two devices (SD an External HDD) and it's working fine i just had to edit the partitions in the os list

**Reply by Jambul46** [reply]
Also i want to have SD always boot first

**Reply by lurch** [reply]
> and it's working fine

Could you please provide more details about your setup, and what it is you'd like to achieve? It's not clear what you're asking for...

**Reply by Jambul46** [reply]
I have SD and HDD conected all time SD is for pin and Kali,RetroPie and HDD is for RPIOS and i would like to add option to boot from disk that is not listed in os.json (if i remember the name correctly) so when connected it would somehow recognize it's Pi bootable and automaticaly pop up in boot list as drive name or something

**Reply by procount** [reply]
Even if PINN detected a bootable external hd that you have just plugged in, it cannot boot it. It can only boot a boot partition from the pinn drive.

If pinn is on sdcard and you use it to install an os onto a usb device, then the new boot partition is still stored on the sdcard so pinn can boot it, even though any other partitions will be stored on the usb.

The only way you can boot an additional device is to make that the boot device in preference to pinn usung bootorder as above.

I'm afraid that's just the way Pi's built on firmware works.

**Reply by Jambul46** [reply]
I was able to boot from external boot partition just fine

**Reply by procount** [reply]
Can you post your modified /settings/installed_os.jso file so we can understand what you're doing?

**Reply by Jambul46** [reply]
Iam not home currently so it will have to wait till tomorrow
